libbeat/template/pipeline.go

```diff
@@ -0,0 +1,154 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package template
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/elastic/beats/libbeat/common"
+	commonP "github.com/elastic/beats/libbeat/common/pipeline"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+var (
+	esIngestPipelinePath = "/_ingest/pipeline/"
+
+	beatsDefaultPipeline = common.MapStr{
+		"description": "Beats default pipeline for enrichment with ingest timestamp, GeoIP, and ASN",
+		"processors": []common.MapStr{
+			common.MapStr{
+				"geoip": common.MapStr{
+					"if":             "ctx.source?.geo == null",
+					"field":          "source.ip",
+					"target_field":   "source.geo",
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"geoip": common.MapStr{
+					"if":             "ctx.destination?.geo == null",
+					"field":          "destination.ip",
+					"target_field":   "destination.geo",
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"geoip": common.MapStr{
+					"if":            "ctx.source?.as == null",
+					"field":         "source.ip",
+					"target_field":  "source.as",
+					"database_file": "GeoLite2-ASN.mmdb",
+					"properties": []string{
+						"asn",
+						"organization_name",
+					},
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"geoip": common.MapStr{
+					"if":            "ctx.destination?.as == null",
+					"field":         "destination.ip",
+					"target_field":  "destination.as",
+					"database_file": "GeoLite2-ASN.mmdb",
+					"properties": []string{
+						"asn",
+						"organization_name",
+					},
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"rename": common.MapStr{
+					"field":          "source.as.asn",
+					"target_field":   "source.as.number",
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"rename": common.MapStr{
+					"field":          "source.as.organization_name",
+					"target_field":   "source.as.organization.name",
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"rename": common.MapStr{
+					"field":          "destination.as.asn",
+					"target_field":   "destination.as.number",
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"rename": common.MapStr{
+					"field":          "destination.as.organization_name",
+					"target_field":   "destination.as.organization.name",
+					"ignore_missing": true,
+				},
+			},
+			common.MapStr{
+				"set": common.MapStr{
+					"field": "event.ingested",
+					"value": "{{_ingest.timestamp}}",
+				},
```

Comment:
This is cool. I wonder if `event.ingested` should be in ECS. CC @MikePaquette @webmat

Reply:
Already on it: https://github.com/elastic/ecs/pull/582 :-)